src/parquet/column/properties.h

```diff
@@ -139,21 +155,23 @@ class WriterProperties {
       return this;
     }
 
-    Builder* compression(const std::shared_ptr<schema::ColumnPath>& path,
-                         Compression::type codec) {
+    Builder* compression(
+        const std::shared_ptr<schema::ColumnPath>& path, Compression::type codec) {
       return this->compression(path->ToDotString(), codec);
     }
 
     std::shared_ptr<WriterProperties> build() {
-      return std::shared_ptr<WriterProperties>(new WriterProperties(
-          allocator_, dictionary_enabled_, dictionary_pagesize_,
+      return std::shared_ptr<WriterProperties>(new WriterProperties(allocator_,
+          dictionary_enabled_, dictionary_pagesize_, default_encoding_, encodings_,
           pagesize_, version_, default_codec_, codecs_));
     }
 
    private:
     MemoryAllocator* allocator_;
     bool dictionary_enabled_;
     int64_t dictionary_pagesize_;
+    Encoding::type default_encoding_;
+    std::unordered_map<std::string, Encoding::type> encodings_;
```

Comment:
Or perhaps refactor these pairs of variables into instances of `PerColumnProperties<Encoding::type/Compression::type>` to avoid duplicating comments.